qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -742,39 +658,164 @@ def risefall_sigma_ratio(self):
         return (self.duration - self.width) / (2.0 * self.sigma)
 
 
+class GaussianSquareEcho(SymbolicPulse):
+    """An echoed Gaussian square pulse.
+    Exactly one of the ``risefall_sigma_ratio`` and ``width`` parameters has to be specified.
+    If ``risefall_sigma_ratio`` is not None and ``width`` is None:
+    .. math::
+        \\text{risefall} &= \\text{risefall_sigma_ratio} \\times \\text{sigma}\\\\
+        \\text{width} &= \\text{duration} - 2 \\times \\text{risefall}
+    If ``width`` is not None and ``risefall_sigma_ratio`` is None:
+    .. math:: \\text{risefall} = \\frac{\\text{duration} - \\text{width}}{2}
+    """
+
+    def __init__(
+        self,
+        duration: Union[int, ParameterExpression],
+        amp: Union[complex, ParameterExpression],
+        sigma: Union[float, ParameterExpression],
+        width: Optional[Union[float, ParameterExpression]] = None,
+        echo_angle: Optional[Union[float, ParameterExpression]] = 0,
+        active_amp: Optional[Union[complex, ParameterExpression]] = 0,
+        active_angle: Optional[Union[float, ParameterExpression]] = 0,
+        risefall_sigma_ratio: Optional[Union[float, ParameterExpression]] = None,
+        name: Optional[str] = None,
+        limit_amplitude: Optional[bool] = None,
+    ):
+        """Create new pulse instance.
+        Args:
+            duration: Pulse length in terms of the sampling period `dt`.
+            amp: The amplitude of the Gaussian and of the square pulse.
+            sigma: A measure of how wide or narrow the Gaussian risefall is; see the class
+                   docstring for more details.
+            width: The duration of the embedded square pulse.
+            echo_angle: Initial phase of the echo pulse.
```

Comment:
Actually I like the phasor representation, but for now Qiskit conforms to the complex representation rather than having amp and angle.

Reply:
Got it! Ok, changed to complex amplitudes only